### PR TITLE
New Routes for create and index for Rents

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
New routes "create" and "index" for users/:id/rents.
### ScreenShoots
image
image
### Trello
https://trello.com/c/EUg3c7j5/29-crear-endpoint-de-creacion-de-rent-e-index

### CODE REVIEW

File: app/controllers/api/v1/rents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Api
+  module V1
+    class RentsController < ApplicationController
+      before_action :authenticate_user!
+      def index
+        render_paginated Rent.where(user_id: params[:user_id]), each_serializer: RentSerializer
```
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb]: `render_paginated user.rents, each_serializer: RentSerializer`, besides being easier to understand will be faster, rails uses index.

File: app/controllers/api/v1/rents_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      end
+
+      def create
```
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb:9]:
```
new_rent = user.rents.new(create_rent_params)
if new_rent.save
  render json: new_rent, serializer: RentSerializer, status: :created
else
  render json: { errors: new_rent.errors }, status: :unprocessable_entity
end
```

Comment (facucompe) [reply app/controllers/api/v1/rents_controller.rb:9]: `create_rent_params` is a method that require and permit params. Check this part of rails guides [https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters](url)

**Review by facucompe — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/rents_controller.rb (hunk 1 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb]: Done!

File: app/controllers/api/v1/rents_controller.rb (hunk 2 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb:9]: i got it.

Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb:9]: Done! (Now it's true!)

File: app/controllers/api/v1/rents_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module Api
+  module V1
+    class RentsController < ApplicationController
+      before_action :authenticate_user!
+      def index
+        user = User.find(params[:user_id])
```
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb]: There should be a method of devise token auth that gives you the current user.

File: app/controllers/api/v1/rents_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        render_paginated user.rents, each_serializer: RentSerializer
+      end
+
+      def create
+        user = User.find(params[:user_id])
```
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb]: Same here

**Review by facucompe — CHANGES_REQUESTED** [review]
Nice work, almost there

File: app/controllers/api/v1/rents_controller.rb (hunk 3 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb]: I don't think that it be necessary for the next reason. In Apiary the path include the user id (this way user/:userid/books) so i suppose what any user authenticated can do a request for get rents info of any user.

File: app/controllers/api/v1/rents_controller.rb (hunk 4 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb]: The same reasons that I've exposed previously.

File: app/controllers/api/v1/rents_controller.rb (hunk 3 above)
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb]: One last change after merging. Move this to a method so you don't repeat code
```
def user
  User.find(params[:user_id])
end
```

**Review by facucompe — DISMISSED** [review]

**Review by facucompe — APPROVED** [review]

**Pull request merged**